### DISCUSSION TITLE
Medical Support App/Telegram robot

### DISCUSSION THREAD

**Discussion by parhamsms** [WeAreMahsaAmini/Core#43] [Ideas]
Purpose: 
To provide free medical services for compatriots living in Iran who cannot access medical services

Patient group: 
Eye injury, head injury, skin injury, abdominal injury, broken bones, wound, penetrating trauma, young patients with heart problems, psychological injury

Required specialties: 
general physician, emergency medicine, ophthalmologist, neurosurgeon, orthopedic surgeon, general surgeon, plastic surgeon, vascular surgeon, ENT specialist, cardiologist, internist, radiologist, psychiatrist, pharmacist, IT engineer, software engineer, Network security engineer, graphic artist

Criteria for attracting doctors:
- Non-affiliation with military organizations
- Lack of activity in government or military centers of Iran
- History of people's support in the last 5 years

How to attract doctors:
Doctors who have the above are added to the doctors by a physician colleague. All doctors are anonymous and no one knows the identity of other doctors. Their Telegram ID or any app is determined based on their specialty and does not contain identifying information of that person (no name, no photo). Considering that the doctors do not know each other's identity, to ensure that they have the expertise in question, two trusted experts in that field determine their scientific qualifications.

[pezeshkan iran.pdf](https://github.com/WeAreMahsaAmini/Core/files/9660451/pezeshkan.iran.pdf)

**Comment by parhamsms** [comment]
I can create a medical team of Iranian doctors from all over the world.
I want you to design an app/ telegram robot that:
- Have high security and be impenetrable
- The information of patients and doctors should be completely confidential
- For greater security, all information should be deleted automatically

**Comment by syavash** [comment]
@parhamsms This is an amazing idea. We can associate a section in WeAreMahsaAmini.com website and our under development Telegram bot.

**Reply by sorsaffari** [reply]
Hi @syavash. Kudos to you for taking this initiative.

I'm not sure if trying to incorporate everything into a few projects (such as the Weare Mahsaamini.com website or a single Telegram bot) is the best idea.

This will slow things down significantly due to bureaucracy, having to communicate asynchronously with people across timezones, technical blockers and dependencies, etc.

Instead, I'd like to suggest the core team create guides to help the community validate their ideas, identify roles, recruit and collaborate, as quickly as possible in the leanest way possible. There should be a team dedicated to not working on any single project, but rather  ensuring that the right projects get picked up by the right teams serving a clear and important purpose. We need a "success" team.

This movement is too fast and initiatives such as this should adapt to the high pace.

**Reply by syavash** [reply]
@sorsaffari I absolutely agree.

Here is the to start a new project or contribute to one: https://github.com/WeAreMahsaAmini/Core/blob/main/CONTRIBUTING.md

Feel free to open a PR to improve it.

**Comment by kaveh6202** [comment]


**Reply by kaveh6** [reply]
@atriaa could you remove your reply as the comment has been removed .

**Comment by atriaa** [comment]
Not sure how doctors and providers have access to medical supplies in Iran, but I imagine not all of them have an infinite amount of supplies. Maybe we can create a donation section where people in Iran (or outside, if we find a way) are able to donate money or medical supplies for doctors.

**Comment by kaveh6** [comment]
I can take care of the telegram part . I already have a project for telegram bot . it's just a matter of implementing steps .

**Comment by kaveh6** [comment]
from my pov , there would be 2 kind of services , one that doesnt require any personal information , an injured person can simply join as anonymous and seek medical information or even have a chat with a doctor about an issue . On the other hand , some injuries would require physical attention and doctor and injured person should meet ... for approving people for this kind of service , i can only think of a referral only Authentication system , and whoever with enough "trust worthy" point can refer another person to join. medical staff can only join through a person with same role and enough "trust worthy" point .
Having a database like this is another issue since it comes with a lot of security concerns that needs to be taken care of .

Another issue is for when a "joined" or "trust worthy" person's phone compromised and some one else get access to it . We could think of some mechanisms to prevent this; on the top of my head, every one should have a unique permanent code and a temporary code(google auth) for seeking such services . Or even send a voice and repeat the permanent code to confirm his/her authenticity every time they seek such service . And also a fake code to be used in compromised situation under pressure .

**Reply by parhamsms** [reply]
Good tips

**Comment by Razz19** [comment]
Hey, I was thinking about the same idea, So happy that you opened this discussion
I know [Dr.Kay](https://www.instagram.com/_dr.kay_/) on Instagram is doing the same thing and I think we can reach him to help us with the features, and maybe he can help us find trustworthy medical experts. 
IMO it will be better if @syavash reaches him because he started the project, and definitely can explain it better to others

**Reply by parhamsms** [reply]
Dr. Kevan is doing this alone on Instagram
He also has to take help from Instagram due to the lack of such an application
We have many similar doctors and we will definitely use their opinions and experiences

**Comment by parhamsms** [comment]
I will explain further:

We want a program/robot that can take the medical history of the patient in the form of text, photo, or video that does not contain the patient's identity information. This program or robot transmits medical information to the triage group. A triage group is a place where several general practitioners are always present and review the medical history and information of the patients. At this time, we have 3 types of patients.

- Level 1 patients:
 Patients who need urgent medical intervention should go to the hospital as soon as possible. These patients cannot be treated remotely and need medical interventions and special hospital care. What the triage doctor does for these patients is pre-hospital care so that the patient can be transferred to the hospital in better conditions and the patient can reach the hospital alive. Also, at this stage, patients can be taught to avoid detection by hospital security.

- Level 2 patients:
 Patients who do not have such an emergency situation, but need specialized treatment, are referred to the specialist of the group. After the triage doctor determines that the patient needs specialized treatment, the patient is referred to the group's specialist. We can have expert Iranian doctors from all over the world and we can also use expert Iranian doctors who live outside of Iran to increase security. The specialist doctor of the group gives the necessary guidance to the patient and the patient's problem is solved. Some patients of this level need medical intervention by a specialist. This is where we can make use of private sector doctors inside Iran. Because in the private sector, it does not matter why the patient was injured. No one discloses the identity of patients and security forces are not present in the private sector. We must create a database of the addresses and phone numbers of specialists in Iran and have a list of specialists by field and city so that we can send this information to patients and patients do not have to go to the hospital due to lack of information.

 - Level 3 patients:
 Patients who do not have an acute problem are guided by a triage doctor and their problem is solved.

[pezeshkan iran 2.pdf](https://github.com/WeAreMahsaAmini/Core/files/9664994/pezeshkan.iran.2.pdf)

**Comment by parhamsms** [comment]
We actually need three parts or panels for this software/robot:

1- Program or robot receiving the patient history
2- part or  panel of triage
3- Part or panel of medical specialists

**Comment by parhamsms** [comment]
I request all programmers to pay more attention to this project. Demand for medical services is extremely high

**Comment by TheTraxex** [comment]
I'm a UX/UI designer and I find this idea extremely helpful so I'm willing to design the UI for this project.

**Comment by vafaei-ar** [comment]
I can help with data organization/potential AI needs.

**Comment by delbarsharghi** [comment]
This is a great idea. I want to be a part of this project. Maybe in the statistical parts or advertising!

**Comment by babooloo** [comment]
Great idea. I'm AI/ML developer with 10years experience in software dev. 
I don't have Telegram specific experience, but if someone with that experience takes the lead, I'll be more than happy to contribute.

@kaveh6 have you made progress on this?

**Comment by parhamsms** [comment]
Dear @syavash, This is a high-priority necessity. Please create a group as soon as possible so that software experts can work on it

**Comment by parhamsms** [comment]
@kaveh6202 @babooloo @delbarsharghi @vafaei-ar @TheTraxex, please coordinate with each other

**Reply by vafaei-ar** [reply]
Any suggestions about how to coordinate?

**Reply by delbarsharghi** [reply]
@syavash jan, what is your idea? If there is a group we can join or I can create a group and start working on it!

**Comment by syavash** [comment]
I just created a channel on Discord for this called #medical-support.

I feel we should be able to build an MVP version of this using a ticketing system like Drift, Intercom, or Zendesk.

Based on today's discussion, it seems one of the requirements for this is to have someone always on call for this and it should be anonymous. If we want to create a Telegram bot from scratch there are two problems:

1. Telegram is filtered in Iran and it requires a proxy. Also, it won't be anonymous.
2. We need to program the rotation and assignment, etc.

Both of the above problems are already figured out in any ticketing system. It will allow chat bots on a website and also it can work through email if the website gets filtered (won't be realtime though.)

We can also look at the stats about how fast Doctors could answer and what's the response time.

What are your thoughts?

**Reply by syavash** [reply]
CC @delbarsharghi @parhamsms

**Reply by delbarsharghi** [reply]
Thank you @syavash jan,
How can we join in Discord? Could you please share the link?

**Reply by syavash** [reply]
@delbarsharghi Here is the link: https://discord.gg/M9yd9S6v

**Reply by delbarsharghi** [reply]
@syavash, Thank you for sharing!

**Reply by parhamsms** [reply]
@syavash 
Thank you for your follow up

In Iran, all messaging software is blocked, although some software, such as Skype and Google Meet, still works.
Many people access international messengers through anti-filters, VPNs, proxies, etc. 

Although we cannot respond to all patients with this method, we can treat many of them.
For the problem of security and anonymity, the Telegram bot can be designed in such a way that it uses the secret chat feature and encrypted end-to-end.

We have thousands of doctors around the world ready to help, and at any hour of the day or night, we'll have dozens of doctors prepared to respond.
I am in contact with many doctors and can create a great medical team.
